### PR TITLE
fix: verifier rewards

### PR DESCRIPTION
**Pull request by renlulu**

### CODE REVIEW

**Review by ansnunez — APPROVED** [review]

**Review by bb111189 — APPROVED** [review]

File: contracts/gzil.scilla [hunk 1]
```diff
@@ -59,9 +59,11 @@ contract GZIL
   symbol: String,
   decimals: Uint32,
   init_supply: Uint128,
-  end_block: BNum
+  during_block: Uint128
```
Comment (bb111189) [inline contracts/gzil.scilla:62]: `during block` abit confusing

maybe  
`num_minting_blocks`

**Review by bb111189 — CHANGES_REQUESTED** [review]
I still feel we need some buffer before activation of minting. This is to prevent 
- people from front-running while the contract is being setup
- sufficient time for us to setup the contract and enroll ssn

I suggest either of the following
- a transition to activate minting. activateMining(uint128 blocknum) 
- or hard code a delayed activation block number

File: contracts/gzil.scilla (hunk 1 above)
Comment (renlulu) [reply contracts/gzil.scilla:62]: OK!

File: contracts/ssnlist.scilla [hunk 2]
```diff
@@ -475,6 +481,11 @@ procedure UpdateStakeReward(entry: SsnRewardShare)
         new_rewards_tmp = builtin mul stake_amt cycle_reward;
         new_rewards = builtin div new_rewards_tmp total_stake_amt;
 
+        (* Substract from verifier_reward, scilla would help us with underflow exception which is always should not happen *)
```
Comment (AmritKumar) [inline contracts/ssnlist.scilla:484]:
```suggestion
        (* Substract from verifier_reward, scilla would help us with underflow exception which should not happen *)
```

**Review by AmritKumar — COMMENTED** [review]
1. Please update README accordingly. 
2. The name of the field `verifier_reward` is a bit misleading.
3. Temporary field is being used which is not very clean.

File: contracts/gzil.scilla [hunk 3]
```diff
@@ -161,23 +171,48 @@ transition ChangeMinter(new_minter: ByStr20)
...
 
+transition ActivateMining(blocknum: Uint128)
+  IsOwner _sender;
+  active <- is_active;
+  match active with
+  | True =>
+    err = CodeAlreadyActive;
+    ThrowError err
+  | False =>
+    end_block_amt = builtin badd _creation_block blocknum;
```
Comment (bb111189) [inline contracts/gzil.scilla:182]: should it be current block number instead of the creation block number?

**Review by bb111189 — APPROVED** [review]

**Review by AmritKumar — APPROVED** [review]

**Pull request merged**